internal/handlers/pg/pgdb/query_test.go

```diff
@@ -164,3 +167,112 @@ func TestQueryDocuments(t *testing.T) {
 		require.NoError(t, tx.Commit(ctx))
 	})
 }
+
+func TestGetDocuments(t *testing.T) {
+	t.Parallel()
+
+	ctx := testutil.Ctx(t)
+
+	pool := getPool(ctx, t)
+	databaseName := testutil.DatabaseName(t)
+	collectionName := testutil.CollectionName(t)
+	setupDatabase(ctx, t, pool, databaseName)
+
+	err := pool.InTransaction(ctx, func(tx pgx.Tx) error {
+		return CreateDatabaseIfNotExists(ctx, tx, databaseName)
+	})
+	require.NoError(t, err)
+
+	cases := []struct {
+		name       string
+		collection string
+		documents  []*types.Document
+	}{{
+		name:       "empty",
+		collection: collectionName + "_empty",
+		documents:  []*types.Document{},
+	}, {
+		name:       "one",
+		collection: collectionName + "_one",
+		documents:  []*types.Document{must.NotFail(types.NewDocument("_id", "foo", "id", "1"))},
+	}, {
+		name:       "two",
+		collection: collectionName + "_two",
+		documents: []*types.Document{
+			must.NotFail(types.NewDocument("_id", "foo", "id", "1")),
+			must.NotFail(types.NewDocument("_id", "foo", "id", "2")),
+		},
+	}, {
+		name:       "three",
+		collection: collectionName + "_three",
+		documents: []*types.Document{
+			must.NotFail(types.NewDocument("_id", "foo", "id", "1")),
+			must.NotFail(types.NewDocument("_id", "foo", "id", "2")),
+			must.NotFail(types.NewDocument("_id", "foo", "id", "3")),
+		},
+	}}
+
+	for _, tc := range cases {
```

Comment:
I'm not sure it is worth it to make this test table-driven: there is no difference between cases "two" and "three", not much difference with case "one", the test has a lot of branches, and we don't even test the case when we break from the iteration with not all rows consumed.

I think, each test should be a separate special (not really) case like "non-existing-collection" below

Reply:
fixed